akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl.fusing
+
+import akka.stream._
+import akka.stream.stage._
+import akka.stream.scaladsl._
+import akka.stream.actor.ActorSubscriberMessage
+import akka.stream.actor.ActorSubscriberMessage._
+import akka.stream.actor.ActorPublisherMessage
+import akka.stream.actor.ActorPublisherMessage._
+import scala.concurrent.forkjoin.ThreadLocalRandom
+
+import java.{ util ⇒ ju }
+
+class FlattenMerge[T, M](breadth: Int) extends GraphStage[FlowShape[Source[T, M], T]] {
```

Comment:
Do we really want a dynamic interleave? I thought that was only for static topologies …